src/main/java/qna/domain/Question.java

```diff
@@ -74,8 +89,49 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public void setDeleted(final boolean deleted) {
-        this.deleted = deleted;
+    private void deleted() {
+        this.deleted = true;
+    }
+
+    public int countOfAnswer() {
+        return answers.size();
+    }
+
+    public List<DeleteHistory> delete(final User user) throws CannotDeleteException {
+        final List<DeleteHistory> deleteHistories = Collections.unmodifiableList(histories(validUser(user)));
+        deleted();
+
+        return deleteHistories;
```

Comment:
`인라인`으로 한줄이 조금 길어졌는데요. 아래처럼 해보면 어떨까요?

```suggestion
        final List<DeleteHistory> deleteHistories = histories(validUser(user));
        deleted();

        return Collections.unmodifiableList(deleteHistories);
```

Reply:
이 부분은 좀 찜찜했는데 좀 더 고민해보겠습니다 👌